aten/src/ATen/NestedTensorImpl.h

```diff
@@ -63,16 +73,26 @@ struct TORCH_API NestedTensorImpl : public c10::TensorImpl {
         " is irregular and does not have a size.");
     return *optional_size;
   }
-
+  /**
+   * Return a view of the nested tensor as a 1 dimensional contiguous tensor.
+   *
+   * The buffer tensor created by this function shares the same storage_impl as
+   * the original nested tensor, and therefore can be seen as a view.
+   *
+   * @return A newly constructed view tensor
+   */
   at::Tensor get_buffer() const {
-    auto buffer_key_set_ = c10::DispatchKeySet{c10::DispatchKey::Dense} |
-        c10::DispatchKeySet{this->key_set_.highestBackendKey()};
+    auto buffer_key_set_ = generate_buffer_key_set();
     auto buffer_tensor_impl = c10::make_intrusive<TensorImpl>(
-        Storage(storage_), buffer_key_set_, data_type_);
+        c10::TensorImpl::VIEW, Storage(storage_), buffer_key_set_, data_type_);
     buffer_tensor_impl->set_sizes_contiguous(c10::makeArrayRef(buffer_size_));
     return Tensor(buffer_tensor_impl);
   }
 
+  int64_t get_buffer_size() const {
+    return buffer_size_;
```

Comment:
Why not `storage_.size()` ? (or whatever is the function name on storage)

Reply:
AFAI storage_.size() or equivalent does not exist. The closest thing that could be done is somethign like: `storage.nbytes()//item_size(dtype)`

Reply:
Sure that sounds better as it avoid having to store yet another field that you have to keep in sync manually no?

Reply:
Okay i'll replace with that